lib/nrf_modem_lib/Kconfig

```diff
@@ -3,31 +3,26 @@
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
 
-menu "BSD Library for nrf91"
-
-config BSD_LIBRARY
-	bool
-	prompt "Use BSD Socket library for IP/TLS/DTLS"
-	imply FPU
-	imply FPU_SHARING
-	select NET_SOCKETS_POSIX_NAMES
-	imply NET_SOCKETS_OFFLOAD
+menuconfig NRF_MODEM_LIB
```

Comment:
This change has be through quite a few peoples that has agreed to the change before this PR was created, but unfortunately it has not been through you before. This move has been planned for some time, and we are currently in a state where we are holding back libmodem production a bit while waiting for this and the nrfxlib PRs to be merged. The goal was to merge this to master early in the NCS 1.4.0 -> NCS 1.5.0 window to give the organization some time to catch up with the change.

If you are saying that we should not do it now we have kind of a problem. We have a fairly big queue of features and bug fixes that is waiting (in the new repo) for this test release to go thought, and going back to the old BSD library would create a fair amount of double work.

Reply:
that is very unfortunate, let me suggest that we keep the existing symbols and deprecate them instead.
I have made a suggestion here: 
https://github.com/lemrey/sdk-nrf/pull/9

Please add any other rename symbols in same manner.

If the deprecated symbol is selected, then we will automatically select the corresponding `NRF_MODEM` symbol.

This will look like:


it can then be decided later if / how we want to inform the user that he is using a deprecated symbol and which symbol he should be using instead.

Reply:
Seems like a good idea, thanks.

Reply:
@tejlmand I have updated the PR based on your suggestion, thank you.

Reply:
@tejlmand @lemrey Is this the best approach? Wouldn't it be better to then just keep bsdlib around instead, we could deprecate bsdlib at a later stage. We don't do any more work on bsdlib, and put a deprecated stamp on the kconfig parameters and and in the change log. And the two libraries would also have to be mutual exclusive.

Isn't this what we asked @carlescufi about earlier lemrey? Because @bjorn-tore-taraldsen was worried about a hard removal of bsdlib, and though we should keep bsdlib around for some time.

Reply:
We just did a rename here, so we agreed that that did not warrant having to keep the old library in the repo.
It is okay to provide a set of compatible KConfig to smooth the transition of those, at least, since it's not much work anyway but we don't need to keep the old lib around.

Reply:
ok